library/datasort.c

```diff
@@ -576,6 +643,17 @@ static int datasort_split(struct datasort_cfg *dcfg)
 	for (n = 0; n < dcfg->bctl_cnt; ++n) {
 		assert(dcfg->bctl[n] != NULL);
 
+		// If base is already sorted we don't need split it into chunks and sort it.
+		// Instead of that we can just iterate it over in the merge phase.
+		if (datasort_base_is_sorted(dcfg->bctl[n])) {
```

Comment:
(not mandatory) `dcfg->bctl[n]` occurs 3 times in the loop body, we can use some proxy variable for it (and pass it to `datasort_add_view_chunk` instead of `n`).